ramips: Add support for Sercomm S1500 clones

MaxS0niX opened the pull request:
This PR adds support for wireless routers:
OEM/ODM Serсomm S1500 AWI - [Beeline SmartBox PRO](https://openwrt.org/inbox/toh/beeline/smartbox_pro)
OEM/ODM Serсomm S1500 BUC - [WiFire S1500.NBN](https://openwrt.org/toh/wifire/s1500_nbn)

- [Specification](https://forum.openwrt.org/t/adding-support-for-sercomm-s1500-clones-beeline-smartbox-pro-wifire-s1500-nbn/110065#general-specification-2)
- [MAC addresses as verified by OEM firmware](https://forum.openwrt.org/t/adding-support-for-sercomm-s1500-clones-beeline-smartbox-pro-wifire-s1500-nbn/110065/15?u=maxs0nix)

### Installation
* Web UI Update
  1. Rename the firmware so that there are no dots in the name, except
  	for the dot in the ".img" file type. Example "openwrt-factory.img".
  2. Upload and update the firmware via the original web interface.

Remark: You might need make the 2rd step twice if your running firmware
is booted from the Slot 1 (Sercomm0 bootflag). The stock firmware
reverses the bootflag (Sercomm0 / Sercomm1) on each firmware update. 

#### Discussion forum topic:
* [Adding support for Sercomm S1500 clones (Beeline SmartBox Pro, WiFire S1500.NBN)](https://forum.openwrt.org/t/adding-support-for-sercomm-s1500-clones-beeline-smartbox-pro-wifire-s1500-nbn/110065)
##### Related devices (SerComm OEM/ODM):
* Sercomm S1500 - #4770 
  * [Beeline SmartBox PRO](https://wikidevi.wi-cat.ru/Beeline_SmartBox_Pro)
  * [WiFire S1500.NBN](https://wikidevi.wi-cat.ru/WiFire_S1500.NBN)
* Sercomm S2
  * [Beeline SmartBox GIGA](https://forum.openwrt.org/t/add-support-for-beeline-smartbox-giga-on-stock-u-boot/99390) - :heavy_check_mark: #4195
* Sercomm S3
  * [Beeline SmartBox TURBO+](https://forum.openwrt.org/t/add-support-for-beeline-smartbox-turbo/99635) - #4108
  * [Beeline SmartBox TURBO](https://forum.openwrt.org/t/add-support-for-beeline-smartbox-turbo/99635/37) - :heavy_check_mark: #10118
  * [Etisalat Sercomm S3](https://forum.openwrt.org/t/add-support-for-sercomm-s3-on-stock-uboot/95970)
  * Rostelecom RT-SF-1
  * Rostelecom RT-FE-1

adschm commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -213,6 +262,38 @@ define Device/asus_rt-n56u-b1
+  IMAGE/kernel.bin := append-ubi | sercomm-tag-header-kernel 0x1700100 0x1f00000
+  IMAGE/rootfs.bin := append-ubi | check-size
+  IMAGE/factory.img := append-ubi | sercomm-tag-factory-img-pro 0x1700100 0x1f00000 0x1b00100 0x3d00000
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
```
What do we need all these images for? Commit message only mentions sysupgrade and initramfs.

MaxS0niX replied:
Left for the duration of the tests. Then leave only factory & sysupgrade?

adschm commented on target/linux/ramips/dts/mt7621_beeline_smartbox-pro.dts:
```diff
@@ -0,0 +1,332 @@
+
+		nvmem-cells = <&macaddr_factory_21000>;
+		nvmem-cell-names = "mac-address";
+		mac-address-increment = <3>;
```
Please add a MAC address overview to the commit message.
https://openwrt.org/docs/guide-developer/device-support-policies#mac_addresses

MaxS0niX replied:
Added to commit. Also described here.
https://forum.openwrt.org/t/adding-support-for-sercomm-s1500-clones-beeline-smartbox-pro-wifire-s1500-nbn/110065/15?u=maxs0nix

adschm commented on target/linux/ramips/dts/mt7621_beeline_smartbox-pro.dts:
```diff
@@ -0,0 +1,332 @@
+		led-boot = &status_white;
+		led-failsafe = &wlan2g;
+		led-running = &status_blue;
+		led-upgrade = &wlan5g;
```
If there are status LEDs available, please don't hijack LED with other purposes.

MaxS0niX replied:
Fixed.

adschm commented:
Please don't change you new scripts again in each patch. One commit for providing the _proper_ script, one commit for adding the device.

MaxS0niX commented:
48b2344 - Changes: 
* The repository was updated to the current state of master.
* Correct MAC's.
* Correct bootloader environment.
* ubi-concat

MaxS0niX commented:
cad82b5 - Changes: 
* The repository was updated to the current state of master.
* LED functions moved to 01_leds
* "ROUT<->REP Switch_bt" turns off Wi-Fi
* Used firmware-utils Sercomm

There are several questions:
1) Is it possible to issue a Pull Request to the firmware-utils repository via github?
2) LEDs do not work for Link 1Gb/s event. How to fix?)
3) Is it possible to add to the device in the next v22.03.x release?
4) There is a similar device [WiFire S1500.NBN](https://gitlab.com/openwrt-dev-ru-smartbox/openwrt-mt7621-sercomm-smartbox/-/tree/sercomm_s1500) is it better to add it in this PR with a common dtsi or a separate one?

hauke commented:
> Is it possible to issue a Pull Request to the firmware-utils repository via github?
No, you have to send a patch to the openwrt devel malling list
> Is it possible to add to the device in the next v22.03.x release?
It should first go into master
> There is a similar device [WiFire S1500.NBN](https://gitlab.com/openwrt-dev-ru-smartbox/openwrt-mt7621-sercomm-smartbox/-/tree/sercomm_s1500) is it better to add it in this PR with a common dtsi or a separate one?
If you have a common dtsi file it is probably easier to add it in one pull request, but in different commits.

hauke commented:
This should be fine:
1) Common dtsi
2) dts PRO + changes in mt7621.mk 
3) dts NBN + changes in mt7621.mk

MaxS0niX commented:
> This should be fine:
> 
>     1. Common dtsi
> 
>     2. dts PRO + changes in mt7621.mk
> 
>     3. dts NBN + changes in mt7621.mk

Thanks for the clarification!

MaxS0niX committed:
ramips: Add support for Sercomm S1500 clones

This PR adds support for wireless routers:
OEM/ODM Serсomm S1500 AWI - Beeline SmartBox PRO
OEM/ODM Serсomm S1500 BUC - WiFire S1500.NBN

General specification
-------------
SoC Type: MediaTek MT7621AT (880 MHz, 2 cores)
Switch: MediaTek MT7530 (via SoC MT7621AT)
Wireless 2.4 GHz (MT7602EN): b/g/n, 2x2
Wireless 5 GHz (MT7612EN): a/n/ac, 2x2
Ethernet: 5 ports - 5×GbE (WAN, LAN1-4)
Mini PCIe (via J2 on PCB, Not soldered on the board)
UART: J4 -> GND[], TX, VCC(3.3V), RX
BootLoader: U-Boot SerComm/Mediatek
PCB: AWI00E-1.6

Beeline SmartBox PRO specification
-------------
RAM (Nanya NT5CB128M16FP): 256 MiB
NAND-Flash (ESMT F59L2G81A): 256 MiB
USB ports: 2xUSB2.0
LEDs: Status (white), WPS (blue), 2g (white), 5g (white) + 10 LED Ethernet
Buttons: 2 button (reset, wps), 1 switch button (ROUT<->REP)
Power: 12 VDC, 1.5 A
PCB Sticker: 970AWI0QW00N256SMT Ver. 1.0
CSN: SG15********
MAC LAN: 94:4A:0C:**:**:**
Manufacturer's code: 0AWI0500QW1

WiFire S1500.NBN specification
-------------
RAM (Nanya NT5CC64M16GP): 128 MiB
NAND-Flash (ESMT F59L1G81MA): 128 MiB
USB ports: 1xUSB2.0
LEDs: Status (white), WPS (white), 2g (white), 5g (white) + 10 LED Ethernet
Buttons: 2 button (RESET, WPS)
Power: 12 VDC, 1.0 A
PCB Sticker: 970BUC0RW00N128SMT Ver. 1.0
CSN: MH16********
MAC WAN: E0:60:66:**:**:**
Manufacturer's code: 0BUC0500RW1

MAC addresses PRO as verified by OEM firmware:

use   address   source
LAN   *:23      factory 0x1000 (label)
WAN   *:24      factory $label +1
2g    *:23      factory $label
5g    *:25      factory $label +2

MAC addresses NBN as verified by OEM firmware:

use   address   source
LAN   *:0e      factory 0x1000
WAN   *:0f      LAN +1 (label)
2g    *:0f      LAN +1
5g    *:10      LAN +2

Installation
------------
* Web UI Update
  1. Rename the firmware so that there are no dots in the name, except
  	for the dot in the ".img" file type. Example "openwrt-factory.img".
  2. Upload and update the firmware via the original web interface.

Remark: You might need make the 2rd step twice if your running firmware
is booted from the Slot 1 (Sercomm0 bootflag). The stock firmware
reverses the bootflag (Sercomm0 / Sercomm1) on each firmware update.

Tested-by: Pavel Ivanov <pi635v@gmail.com>
Signed-off-by: Maximilian Weinmann <x1@disroot.org>

neheb commented:
Only thing I see is the lack of GPIO controlled switch LEDs.

Please adjust to be compatible with https://github.com/openwrt/openwrt/pull/10238
